code/modules/mob/living/simple_animal/bot/cleanbot.dm

```diff
@@ -207,21 +207,17 @@
 /mob/living/simple_animal/bot/cleanbot/UnarmedAttack(atom/A)
 	if(istype(A, /obj/effect/decal/cleanable))
 		anchored = TRUE
-		icon_state = "cleanbot-c"
-		visible_message("<span class='notice'>[src] begins to clean up [A].</span>")
 		mode = BOT_CLEANING
-		spawn(50)
-			if(mode == BOT_CLEANING)
-				if(A && isturf(A.loc))
-					var/atom/movable/AM = A
-					if(istype(AM, /obj/effect/decal/cleanable))
-						for(var/obj/effect/decal/cleanable/C in A.loc)
-							qdel(C)
-
-				anchored = FALSE
-				target = null
-			mode = BOT_IDLE
-			icon_state = "cleanbot[on]"
+		if(isturf(A.loc))
+			var/atom/movable/AM = A
+			if(istype(AM, /obj/effect/decal/cleanable))
+				visible_message("<span class='notice'>[src] cleans up [A].</span>")
+				for(var/obj/effect/decal/cleanable/C in A.loc)
+					qdel(C)
+
+		anchored = FALSE
```

Comment:
Why even make the cleanbot anchored in the first place if the process is instant? Same with the mode. Why even change it if there's no effect?